tools/diagnostics/nsdiff

```diff
@@ -0,0 +1,875 @@
+        #       directories can be transformed and scrubbed in parallel.
+        #       https://docs.python.org/3/library/multiprocessing.html
+        #
+        #       2. Files in Non-LevelDB/RocksDB directories can be transformed
+        #          and scrubbed in parallel.
+        for dirName, subdirList, fileList in os.walk(abspath(sin)):
+            if (dirName in self.skipdirs):
+                logger.debug("Skipping dirName: {}".format(dirName))
+                break
+
+            if self.log_level <= logging.INFO:
+                logger.info("dirName: {}".format(dirName))
+
+            # Skip specific directories
+            subdirList[:] = [x for x in subdirList if x not in self.skipdirs]
+            # Skip specific files
+            fileList[:] = [x for x in fileList if x not in self.skipfiles]
+    
+            # Check if all of the leveldb_rocksdb_filenames are found in
+            # fileList set is_leveldb_rocksdb_dir and break if any of the
+            # filenames are not found
+            is_leveldb_rocksdb_dir = True
+            for filename in leveldb_rocksdb_filenames:
+                if filename not in fileList:
+                    is_leveldb_rocksdb_dir = False
+                    break
+    
+            (dirname, filename, name, extension) = self.split_name(dirName)
+            # Do not include recordings in diff
+            if (filename == "recorder" and dirname.endswith(
+                "/{}".format(sname))):
+                subdirList[:] = []
+                continue
+
+            if (is_leveldb_rocksdb_dir):
+                logger.debug("{} is a LevelDB/RocksDB directory!"
+                    .format(dirName))
+
+                # Use "nodename" in place of the node's name when a directory
+                # (dirName in this case) matches the node name (sname)
+                if (sname == filename):
+                    filename = "nodename"
+
+                if (filename.endswith("_transactions")):
+                    logger.debug("{} is a transaction LevelDB/RocksDB instance!"
+                        .format(filename))
+                    # utf-8 - call "getAllTxn()" in ledger.py in the ledger
+                    #         module/package in indy-plenum.
+                    # Create a Ledger domain object from the ledger directory 
+                    # 'filename' takes the form {fileNamePrefix}_transactions
+                    # where fileNamePrefix is one of the following:
+                    #    config, pool, domain
+                    # FileHashStore requires a fileNamePrefix.
+                    fileNamePrefix = filename.split("_")[0]
+                    ledger = Ledger(
+                        CompactMerkleTree(
+                            # The FileHashStore reads
+                            # {fileNamePrefix}_merkleLeaves and
+                            # {fileNamePrefix}_merkleNodes from dataDir
+                            hashStore=FileHashStore(
```

Comment:
FROM ALEX:
RocksDB-based hash store is actually used by default and in production (see initHashStore)